### PR TITLE
[API][BATCH] add PERF_COUNT to support properties

### PR DESCRIPTION
**Pull request by panhaiqi**
### Details:
 - *[API][BATCH] add PERF_COUNT to support properties*

### Tickets:
 - *CVS-130236*

### CODE REVIEW

File: src/plugins/auto_batch/src/plugin.cpp [hunk 1]
```diff
@@ -82,9 +82,14 @@ ov::Any Plugin::get_property(const std::string& name, const ov::AnyMap& argument
...
     } else if (name == ov::supported_properties.name()) {
-        return std::vector<ov::PropertyName>{
-            ov::PropertyName{ov::supported_properties.name(), ov::PropertyMutability::RO},
-            ov::PropertyName{ov::device::full_name.name(), ov::PropertyMutability::RO}};
+        // return supported_configKeys;
+        std::vector<ov::PropertyName> property_name;
+        property_name.push_back(ov::PropertyName{ov::supported_properties.name(), ov::PropertyMutability::RO});
+        property_name.push_back(ov::PropertyName{ov::device::full_name.name(), ov::PropertyMutability::RO});
+        for (auto& it : supported_configKeys) {
+            property_name.push_back(ov::PropertyName{it, ov::PropertyMutability::RO});
```
Comment (peterchen-intel) [inline src/plugins/auto_batch/src/plugin.cpp:90]: We need to add  RW/RO to supported_configKeys, I think the original intention that config means RW.

**Comment by panhaiqi** [comment]
@zhaixuejun1993 please help review, thanks

File: src/plugins/auto_batch/src/plugin.cpp [hunk 2]
```diff
@@ -82,9 +83,14 @@ ov::Any Plugin::get_property(const std::string& name, const ov::AnyMap& argument
             return {it->second};
         }
     } else if (name == ov::supported_properties.name()) {
-        return std::vector<ov::PropertyName>{
-            ov::PropertyName{ov::supported_properties.name(), ov::PropertyMutability::RO},
-            ov::PropertyName{ov::device::full_name.name(), ov::PropertyMutability::RO}};
+        // return supported_configKeys;
+        std::vector<ov::PropertyName> property_name;
+        property_name.push_back(ov::PropertyName{ov::supported_properties.name(), ov::PropertyMutability::RW});
```
Comment (zhaixuejun1993) [inline src/plugins/auto_batch/src/plugin.cpp:88]: I suppose ov::supported_properties &  ov::device::full_name is read only property, only the config keys can be RW.

**Comment by zhaixuejun1993** [comment]
LGTM

File: src/plugins/auto_batch/src/plugin.cpp [hunk 3]
```diff
@@ -82,9 +83,14 @@ ov::Any Plugin::get_property(const std::string& name, const ov::AnyMap& argument
             return {it->second};
         }
     } else if (name == ov::supported_properties.name()) {
-        return std::vector<ov::PropertyName>{
-            ov::PropertyName{ov::supported_properties.name(), ov::PropertyMutability::RO},
-            ov::PropertyName{ov::device::full_name.name(), ov::PropertyMutability::RO}};
+        // return supported_configKeys;
+        std::vector<ov::PropertyName> property_name;
+        property_name.push_back(ov::PropertyName{ov::supported_properties.name(), ov::PropertyMutability::RO});
+        property_name.push_back(ov::PropertyName{ov::device::full_name.name(), ov::PropertyMutability::RO});
```
Comment (yangwang201911) [inline src/plugins/auto_batch/src/plugin.cpp:89]: Should we register the property `ov::supported_properties` and `ov::device::full_name` into `supported_configKeys` in initiation step?

File: src/plugins/auto_batch/src/plugin.cpp [hunk 4]
```diff
@@ -19,9 +19,10 @@
 namespace ov {
 namespace autobatch_plugin {
 
-std::vector<std::string> supported_configKeys = {ov::device::priorities.name(),
-                                                 ov::auto_batch_timeout.name(),
-                                                 ov::enable_profiling.name()};
+std::vector<ov::PropertyName> supported_configKeys = {
+    ov::PropertyName{ov::device::priorities.name(), ov::PropertyMutability::RW},
```
Comment (yangwang201911) [inline src/plugins/auto_batch/src/plugin.cpp:23]:
```suggestion
    ov::PropertyName{ov::device::priorities.name(), ov::PropertyMutability::RW},
    ov::PropertyName{ov::supported_properties.name(), ov::PropertyMutability::RO},
    ov::PropertyName{ov::device::full_name.name(), ov::PropertyMutability::RO},
```

File: src/plugins/auto_batch/src/plugin.cpp [hunk 5]
```diff
@@ -82,9 +83,14 @@ ov::Any Plugin::get_property(const std::string& name, const ov::AnyMap& argument
             return {it->second};
         }
     } else if (name == ov::supported_properties.name()) {
-        return std::vector<ov::PropertyName>{
-            ov::PropertyName{ov::supported_properties.name(), ov::PropertyMutability::RO},
-            ov::PropertyName{ov::device::full_name.name(), ov::PropertyMutability::RO}};
+        // return supported_configKeys;
+        std::vector<ov::PropertyName> property_name;
```
Comment (yangwang201911) [inline src/plugins/auto_batch/src/plugin.cpp:87]:
```suggestion
        std::vector<ov::PropertyName> property_name = supported_properties;
```

File: src/plugins/auto_batch/src/plugin.cpp [hunk 6]
```diff
@@ -19,9 +19,10 @@
 namespace ov {
 namespace autobatch_plugin {
 
-std::vector<std::string> supported_configKeys = {ov::device::priorities.name(),
-                                                 ov::auto_batch_timeout.name(),
-                                                 ov::enable_profiling.name()};
+std::vector<ov::PropertyName> supported_configKeys = {
```
Comment (yangwang201911) [inline src/plugins/auto_batch/src/plugin.cpp:22]:
```suggestion
std::vector<ov::PropertyName> supported_properties = {
```

File: src/plugins/auto_batch/src/plugin.cpp (hunk 3 above)
Comment (yangwang201911) [reply src/plugins/auto_batch/src/plugin.cpp:91]: resolve this comment as this is the original code logic.

**Review by yangwang201911 — APPROVED** [review]

**Review by zhaixuejun1993 — APPROVED** [review]
LGTM

File: src/plugins/auto_batch/src/plugin.cpp [hunk 7]
```diff
@@ -82,9 +83,14 @@ ov::Any Plugin::get_property(const std::string& name, const ov::AnyMap& argument
             return {it->second};
         }
     } else if (name == ov::supported_properties.name()) {
-        return std::vector<ov::PropertyName>{
-            ov::PropertyName{ov::supported_properties.name(), ov::PropertyMutability::RO},
-            ov::PropertyName{ov::device::full_name.name(), ov::PropertyMutability::RO}};
+        // return supported_configKeys;
```
Comment (riverlijunjie) [inline src/plugins/auto_batch/src/plugin.cpp:86]: Please delete this line.

File: src/plugins/auto_batch/src/plugin.cpp [hunk 8]
```diff
@@ -82,9 +83,14 @@ ov::Any Plugin::get_property(const std::string& name, const ov::AnyMap& argument
...
+        std::vector<ov::PropertyName> property_name;
+        property_name.push_back(ov::PropertyName{ov::supported_properties.name(), ov::PropertyMutability::RO});
+        property_name.push_back(ov::PropertyName{ov::device::full_name.name(), ov::PropertyMutability::RO});
+        for (auto& it : supported_configKeys) {
+            property_name.push_back(it);
+        }
+        return property_name;
```
Comment (riverlijunjie) [inline src/plugins/auto_batch/src/plugin.cpp:93]:
```suggestion
        return decltype(ov::supported_properties)::value_type(std::move(property_name));
```

**Review by riverlijunjie — APPROVED** [review]
LGTM

**Review by wangleis — APPROVED** [review]
@panhaiqi  Please update title and description to clarify the goal of this PR, rather than using title of JIRA ticket.

**Pull request merged**